src/services/VirtualFileSystemService.ts

```diff
@@ -1,259 +1,99 @@
-  }
-
-  private findEntryNoFollow(path: string): VirtualFileSystemEntry | undefined {
-    const parentEntry = this.findParentEntry(path);
-    const name = nodePath.basename(path);
-    if (name === undefined) {
-      throw ErrorFactory.newInternalError('no such file or directory');
-    }
-    return parentEntry.children[name];
-  }
-
-  private findParentEntry(path: string): VirtualDirectory {
-    if (this.structure === undefined) {
-      throw ErrorFactory.newInternalError('structure is undefined');
-    }
-
-    const dirName = nodePath.dirname(path);
-    if (dirName === path) {
-      return this.structure;
-    }
-
-    const entry = this.findEntry(dirName);
-    if (entry === undefined || entry.type !== MetadataType.dir) {
-      throw ErrorFactory.newInternalError('is not a directory');
-    }
-    return entry;
+    vol.reset();
   }
 
   async exists(path: string) {
-    const entry = this.findEntry(path);
-    return entry !== undefined ? true : false;
+    try {
+      await vfs.promises.lstat(path);
+      return true;
+    } catch (error) {
+      return false;
+    }
   }
 
   async readDirectory(path: string) {
-    const entry = this.findEntry(path);
-    if (entry === undefined) {
-      throw ErrorFactory.newInternalError('No such file or directory');
-    }
-    if (entry.type !== MetadataType.dir) {
-      throw ErrorFactory.newInternalError('Is not a directory');
-    }
-
-    return _.keys(entry.children);
+    return <Promise<string[]>>vfs.promises.readdir(path);
   }
 
   async readFile(path: string) {
-    const entry = this.findEntry(path);
-    if (entry === undefined) {
-      throw ErrorFactory.newInternalError('No such file or directory');
-    }
-    if (entry.type !== MetadataType.file) {
-      throw ErrorFactory.newInternalError('Is not a file');
-    }
-
-    return entry.data;
+    return <Promise<string>>vfs.promises.readFile(path, 'utf-8');
   }
 
   async writeFile(path: string, content: string) {
-    let entry = this.findEntry(path);
-    if (entry !== undefined && entry.type !== MetadataType.file) {
-      throw ErrorFactory.newInternalError('is not a file');
-    }
-    this.setEntry(path, { type: MetadataType.file, data: content });
+    return vfs.promises.writeFile(path, content);
   }
 
   async createDirectory(path: string) {
-    if (this.findEntryNoFollow(path) !== undefined) {
-      throw ErrorFactory.newInternalError('Dir already exists');
+    if (!(await this.isDirectory(nodePath.dirname(path)))) {
+      throw ErrorFactory.newArgumentError('No such directory');
     }
-    this.setEntry(path, { type: MetadataType.dir, children: {} });
+
+    return vfs.promises.mkdir(path);
   }
 
   async deleteDirectory(path: string) {
-    if (await this.isDirectory(path)) {
-      this.unsetEntry(path);
-    } else {
-      throw ErrorFactory.newInternalError('Is not a directory');
+    const exists = await this.exists(path);
+    if (!exists || (exists && !(await this.isDirectory(path)))) {
```

Comment:
Is it necessary to check, if it exists? If it's not a directory, it can't be deleted even if it exists, or?